File: frontend/src/api/helm.ts
import { apiFetch } from "./client";
import type { HelmRelease } from "../types/helm";

export function listHelmReleases(namespace?: string): Promise<HelmRelease[]> {
  const q = namespace ? `?namespace=${encodeURIComponent(namespace)}` : "";
  return apiFetch(`/helm/releases${q}`);
}

File: backend/app/api/v1/helm.py
from fastapi import APIRouter
from starlette.concurrency import run_in_threadpool

from app.schemas.helm import HelmRelease
from app.services.k8s_service import K8sService

router = APIRouter(prefix="/helm", tags=["helm"])


@router.get("/releases", response_model=list[HelmRelease])
async def list_helm_releases(namespace: str | None = None) -> list[HelmRelease]:
    return await run_in_threadpool(K8sService().list_helm_releases, namespace)
